qiskit/providers/ibmq/ibmqjob.py

```diff
@@ -220,12 +221,11 @@ def result(self, timeout=None, wait=5):
             raise JobError('Invalid job state. The job should be DONE but '
                            'it is {}'.format(str(status)))
 
-        try:
+        if not self._result:
             job_response = self._get_job()
-        except ApiError as error:
-            raise JobError(str(error))
+            self._result = Result.from_dict(job_response['qObjectResult'])
 
-        return Result.from_dict(job_response['qObjectResult'])
+        return self._result
```

Comment:
Can you revise the docstring? It has fallen out of sync with your latest comment.